lib/Passes/SliceManager.cpp

```diff
@@ -57,14 +57,13 @@ void SliceManager::insertClonedSliceIntoFunction(
   auto bb = llvm::BasicBlock::Create(this->mod.get()->getContext(),
                                      "slicebasicblock." + std::to_string(id.id),
                                      targetFunc);
+  llvm::IRBuilder<> builder(bb);
 
-  std::for_each(slice.begin(), slice.end(), [bb](llvm::Instruction *insn) {
-    bb->getInstList().push_back(insn);
-  });
+  std::for_each(slice.begin(), slice.end(),
+                [&builder](llvm::Instruction *insn) { builder.Insert(insn); });
```

Comment:
Sorry for the noise here. This file hasn't been Clang formatted for a while I guess.

This is the main change in this file. We don't have raw access to the instruction list anymore so I need to do it like this.

We can have an `#if` that calls the new API when we're building with LLVM 16, but I believe the `IRBuilder` is pretty lightweight to construct so I think this is fine.